### DISCUSSION TITLE
Adding Custom Cell

### DISCUSSION THREAD

**Discussion by DiamondGit** [glideapps/glide-data-grid#756] [Q&A]
I cannot add custom cells, in my case button-cell, to table. I installed `glide-data-grid-cells` package already and made all necessary things, but error occurs or no effect:

> import { useExtraCells } from "@glideapps/glide-data-grid-cells";
import type { ButtonCell } from "@glideapps/glide-data-grid-cells/dist/ts/cells/button-cell";
...
const cellProps = useExtraCells();
...
const d: ButtonCell = {
    kind: GridCellKind.Custom,
    cursor: "pointer",
    allowOverlay: true,
    copyData: "4",
    readonly: true,
    data: {
        kind: "button-cell",
        backgroundColor: ["transparent", "#6572ffee"],
        color: ["accentColor", "accentFg"],
        borderColor: "#6572ffa0",
        borderRadius: 9,
        title: "View Details",
        onClick: () => window.alert("Button clicked"),
    },
    themeOverride: {
        baseFontStyle: "700 12px",
    },
};
return d;
...
<DataEditor
    {...cellProps}
    columns={tableColumns}
    rows={data.length}
    rowSelect="none"
    columnSelect="none"
    width="100%"
    getCellsForSelection
    smoothScrollX
    smoothScrollY
    getCellContent={getContent}
/>

**Comment by SourceCipher** [comment]
Yeah, same, I followed the usage on their docs and the cell is just empty, not showing anything

**Comment by SourceCipher** [comment]
Ok I find out why it wasnt working. They need to update the docs.. 

Instead of this:

```
const Grid = () => {
    const { drawCell, provideEditor } = useExtraCells();

    return <DataEditor {...rest} drawCell={drawCell} provideEditor={provideEditor} />;
};
```

Use like this:

```
const Grid = () => {
   const { customRenderers } = useExtraCells()

    return <DataEditor {...rest}  customRenderers={customRenderers} />;
};
```
